### ISSUE TITLE
`call-non-callable` false-positive with UPath introduced in 0.0.82

### ISSUE THREAD

**Issue by sinon** [astral-sh/ty#4552]
### Summary

With the recent `ty` release we are seeing a `call-non-callable` false positive when using the `UPath` class from `universal-pathlib`. MRE using the library in questions:

## MRE with library

```python
# /// script
# requires-python = ">=3.12"
# dependencies = [
#     "universal-pathlib==0.3.10",
# ]
# ///
from upath import UPath

path = UPath("/tmp/example")
print(path)
```

Runs fine:
```sh
$ uv run upath_real.py
/tmp/example
```

No errors on `0.0.81` or pyrelfly 1.3.0:
```sh
uvx ty@0.0.81 check upath_real.py 
All checks passed!
uvx pyrefly@1.3.0 check upath_real.py
 INFO 0 diagnostics
```
Diagnostic now emitted on 0.0.82 
```sh
$ uvx ty@0.0.82 check upath_real.py
error[call-non-callable]: Cannot instantiate abstract class `UPath`
   --> upath_real.py:9:8
    |
  9 | path = UPath("/tmp/example")
    |        ^^^^^^^^^^^^^^^^^^^^^ 5 abstract methods are unimplemented, including `_chain`, `_chain_parser` and `_fs_cached`
    |
   ::: <venv>/lib/python3.14/site-packages/upath/core.py:205:9
    |
205 |     def _chain(self, value: Chain) -> None:
    |         ------ `_chain` declared as abstract on superclass `_UPathMixin`
info: Use `--verbose` to see all 5 unimplemented abstract methods
info: The instance-attribute annotation for `_chain` does not override the abstract method

Found 1 diagnostic
```

## MRE with stdlib

NOTE: This reproduction was generated by LLM attempting to replicate the shape of the underlying library, it looks reasonable but I don't have the time/expertise to verify it fully hopefully it's helpful

[Playground](https://play.ty.dev/25a5fa4e-b521-4998-b612-50e26a79ae53)
```py
from abc import abstractmethod
from typing import TYPE_CHECKING


class Base:
    @property
    @abstractmethod
    def value(self) -> int: ...

    @value.setter
    @abstractmethod
    def value(self, v: int) -> None: ...


class Concrete(Base):
    __slots__ = ("value",)

    if TYPE_CHECKING:
        value: int

    def __init__(self) -> None:
        self.value = 1


Concrete()
```

```sh
error[[call-non-callable](https://ty.dev/rules#call-non-callable)]: Cannot instantiate abstract class `Concrete`
  --> .tmp/ty-mre/repro.py:33:1
   |
33 | Concrete()
   | ^^^^^^^^^^ `value` is unimplemented
   |
  ::: .tmp/ty-mre/repro.py:20:9
   |
20 |     def value(self, v: int) -> None: ...
   |         ----- `value` declared as abstract on superclass `Base`
info: The instance-attribute annotation for `value` does not override the abstract method
help: Either assign a value or add `ClassVar` to this declaration
  --> .tmp/ty-mre/repro.py:27:9
   |
27 |         value: int
   |         ----- Instance-attribute declaration

Found 1 diagnostic
```

### Version

0.0.82

**Comment by AlexWaygood** [comment]
Ah interesting. At runtime (in the MRE stdlib), the presence of `value` in `Concrete.__slots__` creates a `types.MemberDescriptorType` instance in the class namespace that overrides the abstract property in the superclass, meaning that `Concrete` is no longer abstract. We should probably model that.

**Comment by AlexWaygood** [comment]
This should also probably be a dedicated error code rather than the very general `call-non-callable`, now that I think about it again, as well.

**Comment by carljm** [comment]
Bug is fixed; filed https://github.com/astral-sh/ty/issues/4556 to track using a separate diagnostic code.
